labgrid/util/agents/udisks2.py

```diff
@@ -83,17 +90,41 @@ def unmount(self, lazy=False):
 
 _devs = {}
 
-def _get_udisks2_dev(devpath):
+def _get_udisks2_dev(devpath, retries):
+    """Try to get the udisks2 device
+
+    Args:
+        devpath (str): Device name
+        retries (int): Number of retries to allow
+
+    Raises:
+        ValueError: Failed to obtain the device (e.g. does not exist)
+    """
     if devpath not in _devs:
-        _devs[devpath] = UDisks2Device(devpath=devpath)
+        dev = UDisks2Device(devpath=devpath)
+        done = False
+        while not done:
+            try:
+                dev._setup()
+                done = True
+            except ValueError as exc:
+                if 'No udisks2 device' not in str(exc) or not retries:
+                    raise
+                retries -= 1
```

Comment:
retries needs to be compared to zero somewhere, doesn't it?

Reply:
Yes, in line 111